Emby.Server.Implementations/Library/LibraryManager.cs

```diff
@@ -2408,6 +2413,36 @@ public bool FillMissingEpisodeNumbersFromPath(Episode episode, bool forceRefresh
                 episodeInfo = new Naming.TV.EpisodeInfo();
             }
 
+            try
+            {
+                var libraryOptions = GetLibraryOptions(episode);
+                if (libraryOptions.EnableEmbeddedEpisodeInfos && string.Equals(episodeInfo.Container, "mp4", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Read from metadata
+                    var mediaInfo = _mediaEncoder.GetMediaInfo(new MediaInfoRequest
+                    {
+                        MediaSource = episode.GetMediaSources(false)[0],
+                        MediaType = DlnaProfileType.Video,
+                        ExtractChapters = false
+                    }, CancellationToken.None).GetAwaiter().GetResult();
+                    if (mediaInfo.ParentIndexNumber > 0) {
+                        episodeInfo.SeasonNumber = mediaInfo.ParentIndexNumber;
+                    }
+
+                    if (mediaInfo.IndexNumber > 0) {
+                        episodeInfo.EpisodeNumber = mediaInfo.IndexNumber;
+                    }
+
+                    if (!string.IsNullOrEmpty(mediaInfo.ShowName)) {
+                        episodeInfo.SeriesName = mediaInfo.ShowName;
+                    }
+                }
+            }
+            catch (Exception ex)
```

Comment:
Any specific exception you are trying to catch?

Reply:
No. If a exception occurs, the code behaves like it was behaving before this PR.

Reply:
Ok